pylint/testutils/primer.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import git
+
+PRIMER_DIRECTORY_PATH = Path(".pylint_primer_tests")
+
+
+class PackageToLint:
+    """Represents data about a package to be tested during primer tests"""
+
+    url: str
+    """URL of the repository to clone"""
+
+    branch: str
+    """Branch of the repository to clone"""
+
+    directories: List[str]
+    """Directories within the repository to run pylint over"""
+
+    commit: Optional[str] = None
+    """Commit hash to pin the repository on"""
+
+    pylint_additional_args: List[str] = []
+    """Arguments to give to pylint"""
+
+    pylintrc_relpath: Optional[str] = None
+    """Path relative to project's main directory to the pylintrc if it exists"""
+
+    def __init__(
+        self,
+        url: str,
+        branch: str,
+        directories: List[str],
+        commit: Optional[str] = None,
+        pylint_additional_args: Optional[List[str]] = None,
+        pylintrc_relpath: Optional[str] = None,
+    ):
+        self.url = url
+        self.branch = branch
+        self.directories = directories
+        self.commit = commit
+        if pylint_additional_args is None:
+            pylint_additional_args = []
+        self.pylint_additional_args = pylint_additional_args
+        self.pylintrc_relpath = pylintrc_relpath
+
+    @property
+    def pylintrc(self) -> Optional[str]:
+        if self.pylintrc_relpath is None:
+            return None
+        return f"{self.clone_directory}/{self.pylintrc_relpath}"
```

Comment:
```suggestion
    @property
    def pylintrc(self) -> Optional[Path]:
        if self.pylintrc_relpath is None:
            return None
        return self.clone_directory / self.pylintrc_relpath
```
I still recommend using `Path` here. It does work with string formatting as `str(Path)` is used.